lib/vanagon/component.rb

```diff
@@ -290,6 +290,13 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      deps = {}
+      deps['version'] = @version if @version
+      deps['ref'] = @options[:ref] if @options[:ref]
+      { name => deps }
+    end
```

Comment:
Actually, this form would work better since not every object will respond to `#empty?`:

```ruby
>> hsh = { 'arugula' => { 'flavorful': true, 'affordable': nil }.delete_if { |k, v| !v } }
=> {
  "arugula" => {
    :flavorful => true
  }
}
>> puts JSON.pretty_generate hsh
{
  "arugula": {
    "flavorful": true
  }
}
=> nil
>> hsh = { 'arugula' => { 'flavorful': true, 'affordable': nil }.delete_if { |k, v| !v } }
```